### PR TITLE
Apis

### PR DESCRIPTION
**Pull request by ErnestaP**
* created md files with APIs use cases for: Get the status of a task and run, Restart a run from a specific task, Restart a run from the beginning, Restart a run from the beginning with specific parameters
* ref: https://github.com/cern-sis/issues-scoap3/issues/80

### CODE REVIEW

File: apis_instruction.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Restart a run from the beginning
+
+1. Restart run from the beggining:
+   - api: `/api/v1/dags/aps_fetch_api/dagRuns`
+   - header
+     - Content-Type : application/json
+     - Accept : application/json
+   - body:
+     `{ "dag_run_id": "string", "logical_date": date_when_we_want_to_start_run, "conf": { } }`
```
Comment (ErnestaP) [inline apis_instruction.md]: format to json

Comment (ErnestaP) [reply apis_instruction.md]: remove unnecessary fields

File: apis_instruction.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+### Set tasks to state 'failed' (for testing)
+
+1. Set tasks state to failed:
+
+   - api: `api/v1/dags/{dag_id}/updateTaskInstancesState`
+   - header
+     - Content-Type : application/json
+     - Accept : application/json
+   - body:
+     ```python
```
Comment (drjova) [inline apis_instruction.md]: it's json 😃 
```suggestion
     ```json
```

File: airflow.cfg [hunk 3]
```diff
@@ -416,14 +416,14 @@ google_key_path =
...
+access_control_allow_headers = origin, content-type, accept
 
 # Specifies the method or methods allowed when accessing the resource.
-access_control_allow_methods =
+access_control_allow_methods = POST, GET, OPTIONS, DELETE
 
 # Indicates whether the response can be shared with requesting code from the given origins.
 # Separate URLs with space.
-access_control_allow_origins =
+access_control_allow_origins = http://localhost:5432
```
Comment (drjova) [inline airflow.cfg]: this seems hardcoded and will not work for containers, maybe just`*` for development?

File: airflow.cfg [hunk 4]
```diff
@@ -1104,4 +1104,7 @@ shard_code_upper_limit = 10000
 shards = 5
 
 # comma separated sensor classes support in smart_sensor.
-sensors_enabled = NamedHivePartitionSensor
\ No newline at end of file
+sensors_enabled = NamedHivePartitionSensor
+
+# [database]
+# sql_alchemy_conn = postgresql+psycopg2://airflow:airflow@localhost:5432/airflow
```
Comment (drjova) [inline airflow.cfg]: ? is this a left over?

File: docker-compose.yaml [hunk 5]
```diff
@@ -59,6 +59,8 @@ x-airflow-common: &airflow-common
     - ./dags:/opt/airflow/dags
     - ./logs:/opt/airflow/logs
     - ./plugins:/opt/airflow/plugins
+    - ./airflow.cfg:/opt/airflow/airflow.cfg
+    - ./webserver_config.py://opt/airflow/webserver_config.py
```
Comment (drjova) [inline docker-compose.yaml]: AFAIS we don't have such file :/

File: airflow.cfg [hunk 6]
```diff
@@ -416,14 +416,14 @@ google_key_path =
...
+access_control_allow_headers = origin, content-type, accept
 
 # Specifies the method or methods allowed when accessing the resource.
-access_control_allow_methods =
+access_control_allow_methods = POST, GET, OPTIONS, DELETE
 
 # Indicates whether the response can be shared with requesting code from the given origins.
 # Separate URLs with space.
-access_control_allow_origins =
+access_control_allow_origins =
```
Comment (drjova) [inline airflow.cfg:426]: if you leave it empty it will allow everything i.e. `*`?

**Pull request merged**